vcsclient/azurerepos.go

```diff
@@ -365,3 +367,84 @@ func (client *AzureReposClient) DownloadFileFromRepo(ctx context.Context, owner,
 func (client *AzureReposClient) GetRepositoryEnvironmentInfo(ctx context.Context, owner, repository, name string) (RepositoryEnvironmentInfo, error) {
 	return RepositoryEnvironmentInfo{}, getUnsupportedInAzureError("get repository environment info")
 }
+
+func (client *AzureReposClient) GetModifiedFiles(ctx context.Context, _, repository, beforeSha, afterSha string) ([]string, error) {
+	if err := validateParametersNotBlank(map[string]string{
+		"repository": repository,
+		"beforeSha":  beforeSha,
+		"afterSha":   afterSha,
+	}); err != nil {
+		return nil, err
+	}
+
+	azureReposGitClient, err := client.buildAzureReposClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var fileNamesList []string
+	fileNamesSet := make(map[string]bool)
```

Comment:
This is not exactly a set, but a map. I'd recommend using the Set data structure from our [gofrog](https://github.com/jfrog/gofrog) library:
https://github.com/jfrog/gofrog/blob/master/datastructures/set.go

Same comment for the other Git providers